store/types/store.go

```diff
@@ -439,3 +447,34 @@ type StoreWithInitialVersion interface {
 	// starting a new chain at an arbitrary height.
 	SetInitialVersion(version int64)
 }
+
+type (
+	PruningOptions  = pruningtypes.PruningOptions
+	PruningStrategy = pruningtypes.PruningStrategy
+)
```

Comment:
```go
type X = Y
```

This creates a type alias. [Type aliases are a niche feature](https://tip.golang.org/doc/go1.9#language) meant to

> support gradual code repair while moving a type between packages

But that's not what this is. What you want here is just a plain old type definition.

```go
type (
type (
  	PruningOptions  pruningtypes.PruningOptions
  	PruningStrategy pruningtypes.PruningStrategy
)
```

Reply:
I was not aware of this difference, thanks for pointing it out.

I looked more into this, and it seems that type aliases are also meant for "allowing large packages to be split into multiple implementation packages with a single top-level exported API" ([Source](https://github.com/golang/proposal/blob/master/design/18130-type-alias.md)) 

That's what I originally meant to do by referencing other examples in the codebase such as this one: https://github.com/osmosis-labs/cosmos-sdk/blob/aff36b2589cc816946150df4b8513ec32fa1c695/types/store.go#L12-L23

However, after looking into this more, I think it is cleaner to import: `"github.com/cosmos/cosmos-sdk/pruning/types"` where necessary, instead of consolidating all types in `"github.com/cosmos/cosmos-sdk/store/types"`

As a result, I removed these. Please let me know what you think